### ISSUE TITLE
Object leak when subclassing Resource twice within the same script

### ISSUE THREAD

**Issue by Rubonnek** [godotengine/godot#40330]
**Godot version:**
3.2 branch commit c2a4ce9380736b65c27fb204c307bdd4d84b43b6

**OS/device including version:**
Arch Linux

**Issue description:**
I was cleaning up my own branch of [this fsm](https://github.com/godot-addons/godot-finite-state-machine) implementation when I noticed a strange object leak.

**Steps to reproduce:**

Simply drop the following script in a brand new project and open and close Godot twice -- you'll get a resource leak that's triggered by the `State` class when extending `Resource`.
```gdscript
extends Resource

class_name StateMachine
var m_dummy : Object

# Note State extends Resource -- leak reported
class State extends Resource:
	var m_state_machine: StateMachine
````

This leak, however, is not reported if we use the following script instead:

```gdscript
extends Resource

class_name StateMachine
var m_dummy : Object

# Now State extends StateMachine which extends Resource -- no leak reported
class State extends StateMachine:
	var m_state_machine: StateMachine
````

**Minimal reproduction project:**
[ResourceLeakTest.zip](https://github.com/godotengine/godot/files/4912699/ResourceLeakTest.zip)

**Comment by Rubonnek** [comment]
There's something else that's strange here -- when running a scene by pressing F6 and closing the window of the running scene, a leak will be reported. If you press F8 on the Editor however, looks like Resource reference count cycles down successfully since no leak gets reported.

**Comment by Rubonnek** [comment]
For reference, this is what I get when running `godot -e --verbose`:

```
Godot Engine v3.2.3.beta.custom_build.5bec48400 - https://godotengine.org
XInput: Refreshing devices.
XInput: No touch devices found.
Detecting GPUs, set DRI_PRIME in the environment to override GPU detection logic.
Only one GPU found, using default.
XcursorGetTheme could not get cursor theme
Using GLES3 video driver
OpenGL ES 3.0 Renderer: GeForce GPU/PCIe/SSE2
PulseAudio: detected 2 channels
PulseAudio: audio buffer frames: 512 calculated latency: 11ms
 
CORE API HASH: 10142890207221811632
EDITOR API HASH: 14959599615661278174
Loading resource: res://default_env.tres
Construct gdnative interface

Destruct gdnative interface

Class 'BulletPhysicsDirectBodyState' is not exposed, skipping.
Class 'BulletPhysicsDirectSpaceState' is not exposed, skipping.
Class 'BulletPhysicsServer' is not exposed, skipping.
Class 'GDScriptNativeClass' is not exposed, skipping.
Class 'IP_Unix' is not exposed, skipping.
Class 'InputDefault' is not exposed, skipping.
Class 'Physics2DDirectBodyStateSW' is not exposed, skipping.
Class 'Physics2DDirectSpaceStateSW' is not exposed, skipping.
Class 'Physics2DServerSW' is not exposed, skipping.
Class 'ResourceImporterOGGVorbis' is not exposed, skipping.
Loading resource: /home/wilson/.config/godot/editor_settings-3.tres
EditorSettings: Load OK!
Loaded builtin certs
EditorSettings: Save OK!
Loading resource: res://Node2D.tscn
Loading resource: res://Node2D.gd
Loading resource: res://state_machine.gd
EditorSettings: Save OK!
ERROR: ~List: Condition "_first != __null" is true.
   At: ./core/self_list.h:112.
ERROR: ~List: Condition "_first != __null" is true.
   At: ./core/self_list.h:112.
WARNING: cleanup: ObjectDB instances leaked at exit (run with --verbose for details).
   At: core/object.cpp:2135.
Leaked instance: GDScript:18106 - Resource path: 
Leaked instance: GDScript:18105 - Resource path: res://state_machine.gd
Leaked instance: GDScriptNativeClass:553
Hint: Leaked instances typically happen when nodes are removed from the scene tree (with `remove_child()`) but not freed (with `free()` or `queue_free()`).
ERROR: clear: Resources still in use at exit (run with --verbose for details).
   At: core/resource.cpp:477.
Resource still in use: res://state_machine.gd (GDScript)
Orphan StringName: res://state_machine.gd
Orphan StringName: GDScriptNativeClass
Orphan StringName: m_dummy
Orphan StringName: State
Orphan StringName: Object
Orphan StringName: m_state_machine
Orphan StringName: Resource
Orphan StringName: _init
Orphan StringName: GDScript
StringName: 9 unclaimed string names at exit.
```

**Comment by Reisflocke** [comment]
I ran into a similar issue last week. (Godot 3.2.2 stable)

I got two scripts.
Each of them has a class_name.
Now I'm referring to class_b in script_a & class_a in script_b
It causes kinda of the same problem

I think it's  just trying to:
1: load script_b in script_a
2: realizes that script_a is refereed to in script_b
3: so it tries to load script _a in script_b
4: then it agian realizes that script_b is refereed to in script_a
x: and the cycle repeats
